operator/controller/src/test/java/io/apicurio/registry/operator/it/OLMITBase.java

```diff
@@ -0,0 +1,124 @@
+    protected static boolean cleanup;
+
+    @BeforeAll
+    public static void beforeAll() throws Exception {
+        setDefaultAwaitilityTimings();
+        namespace = calculateNamespace();
+        client = createK8sClient(namespace);
+        createNamespace(client, namespace);
+        ingressManager = new IngressManager(client, namespace);
+        cleanup = ConfigProvider.getConfig().getValue(CLEANUP, Boolean.class);
+
+        if (client.apiextensions().v1().customResourceDefinitions()
+                .withName("catalogsources.operators.coreos.com").get() == null) {
+            throw new OperatorException("CatalogSource CRD is not available. Please install OLM.");
+        }
+
+        var projectVersion = ConfigProvider.getConfig().getValue(PROJECT_VERSION_PROP, String.class);
+        var projectRoot = ConfigProvider.getConfig().getValue(PROJECT_ROOT_PROP, String.class);
+        var catalogImage = ConfigProvider.getConfig().getValue(CATALOG_IMAGE_PROP, String.class);
+
+        var testDeployDir = Paths.get(projectRoot, "operator/controller/src/test/deploy");
+
+        // Catalog Source
+
+        var catalogSourceRaw = Files.readString(testDeployDir.resolve("catalog/catalog-source.yaml"));
+        catalogSourceRaw = catalogSourceRaw.replace("${PLACEHOLDER_CATALOG_NAMESPACE}", namespace);
+        catalogSourceRaw = catalogSourceRaw.replace("${PLACEHOLDER_CATALOG_IMAGE}", catalogImage);
+        var catalogSource = client.resource(catalogSourceRaw);
+        catalogSource.create();
+
+        await().ignoreExceptions().until(() -> {
+            return client.pods().inNamespace(namespace).list().getItems().stream().filter(
+                    pod -> pod.getMetadata().getName().startsWith("apicurio-registry-operator-catalog"))
+                    .anyMatch(pod -> pod.getStatus().getConditions().stream()
+                            .anyMatch(c -> "Ready".equals(c.getType()) && "True".equals(c.getStatus())));
+        });
+
+        // Operator Group to allow deploying of the operator to the test namespace
+
+        var operatorGroupRaw = Files.readString(testDeployDir.resolve("catalog/operator-group.yaml"));
+        operatorGroupRaw = operatorGroupRaw.replace("${PLACEHOLDER_NAMESPACE}", namespace);
+        var operatorGroup = client.resource(operatorGroupRaw);
+        operatorGroup.create();
+
+        // Subscription
+
+        var subscriptionRaw = Files.readString(testDeployDir.resolve("catalog/subscription.yaml"));
+        subscriptionRaw = subscriptionRaw.replace("${PLACEHOLDER_NAMESPACE}", namespace);
+        subscriptionRaw = subscriptionRaw.replace("${PLACEHOLDER_CATALOG_NAMESPACE}", namespace);
+        subscriptionRaw = subscriptionRaw.replace("${PLACEHOLDER_PACKAGE}",
+                "apicurio-registry.v" + projectVersion.toLowerCase());
+        var subscription = client.resource(subscriptionRaw);
+        subscription.create();
+
+        // Wait for the operator to deploy
```

Comment:
I like how you implemented the logical steps here, connected to another comment above I'd suggest:

let's move this test with the code below "Wait for the operator to deploy" as the actual  `@Test` in a separate sub-module.

Reasoning is:
we are already testing the operator in local and remote and the OLM test only need to test that the OLM integration works. A separate Maven sub-module seems appropriate here.

Reply:
So, consider the deploy check as the actual test, instead of running smoke tests with OLM? I'm OK with this, but since the test will be very simple, let's wait with moving to a different module when it becomes complex enough.

Reply:
> consider the deploy check as the actual test

correct.

> let's wait with moving to a different module

Here is not about complexity, it's about mixing concerns.
The installation of the operator is a concern not needed for most of the tasks, and keeping it in a separate sub-module is a clean way to avoid additional flag and knobs(such as `disableIf`) that should be looked up every time we run the testsuite.

Reply:
done.